src/unity.c

```diff
@@ -1153,6 +1156,54 @@ void UnityAssertEqualMemory(UNITY_INTERNAL_PTR expected,
 }
 
 /*-----------------------------------------------*/
+#ifndef UNITY_EXCLUDE_FILE
+void UnityAssertEqualBinaryFile(const char* expected,
+                                const char* actual,
+                                const char* msg,
+                                const UNITY_LINE_TYPE lineNumber,
+                                const UNITY_FLAGS_T flags)
+{
+
+    FILE* expectedFile = NULL;
+    FILE* actualFile = NULL;
+    UNITY_UINT32 expectedSz, actualSz;
+    UNITY_PTR_ATTRIBUTE unsigned char* ptr_exp = NULL;
+    UNITY_PTR_ATTRIBUTE unsigned char* ptr_act = NULL;
+
+    RETURN_IF_FAIL_OR_IGNORE;
+
+    expectedFile = fopen(expected, "r");
+
+    if (expectedFile == NULL)
+        UNITY_FAIL_AND_BAIL;
+    fseek(expectedFile, 0, SEEK_END);
+    expectedSz = (UNITY_UINT32) ftell(expectedFile);
+    fseek(expectedFile, 0, SEEK_SET);
+    ptr_exp = malloc(expectedSz);
+    fread(ptr_exp, 1, expectedSz, expectedFile);
+    fclose(expectedFile);
+
+    actualFile = fopen(actual, "r");
```

Comment:
Same as above i also changed this to "rb" because of some memory compare problems.